File: src/application/use_cases/status.py
"""Status-related application use cases."""

from __future__ import annotations

from src.application.ports import StatusReader


def list_context_status(reader: StatusReader) -> list[dict[str, object]]:
    return reader.list_context_status()


def validate_context_network(
    context_name: str,
    reader: StatusReader,
) -> dict[str, object]:
    return reader.validate_context_network(context_name)

File: internal/application/usecases/status.go
package usecases

import "github.com/systemframe/k3ctx/internal/domain"

// ListContextStatus returns all active context status entries.
func ListContextStatus(reader domain.StatusReader) ([]map[string]any, error) {
	return reader.ListContextStatus()
}

// ValidateContextNetwork returns the network validation result for a context.
func ValidateContextNetwork(contextName string, reader domain.StatusReader) (map[string]any, error) {
	return reader.ValidateContextNetwork(contextName)
}
